Namespace(batch_size=1000, cpuonly=False, dataset='inaturalist', downloadall=False, end=10000, freeze=True, freeze_idx=25, model='vgg11', num_epochs=1, sequential=False, split_choice='automatic', split_idx=100, start=0, testonly=False, use_intermediate=False)
==> Preparing data..
==> Building model..
Using split index: 100
Freezing the lower layers of the model (vgg11) till index 25
The mode is:  vanilla
Memory occpied: (1874.0, 3.0)
Memory occpied: (1874.0, 3.0)
Memory occpied: (1874.0, 3.0)
Memory occpied: (1874.0, 3.0)
Read 265.7877779006958 MBs for this batch
Streaming inaturalist data took 4.721357107162476 seconds
The mode is:  vanilla

Epoch: 0
Time of next(dataloader) is: 4.790247678756714
Time for copying to cuda: 0.15180635452270508
Read 266.2351598739624 MBs for this batch
Streaming inaturalist data took 5.018665790557861 seconds
Time for forward pass: 15.049292802810669
Time for backpropagation: 0.17213106155395508
GPU memory for training: 22.989273071289062                          

One training iteration takes: 20.375165462493896 seconds
Index: 0
Then, training+dataloading take 20.380879402160645 seconds
The mode is:  vanilla

Epoch: 0
Memory occpied: (1874.0, 11660.0)
Memory occpied: (5054.0, 11660.0)
Memory occpied: (5054.0, 11660.0)
Memory occpied: (5054.0, 11660.0)
Time of next(dataloader) is: 4.847603797912598
Time for copying to cuda: 0.15839672088623047
Exception: Caught RuntimeError in replica 0 on device 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/parallel/parallel_apply.py", line 61, in _worker
    output = module(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torchvision/models/vgg.py", line 66, in forward
    x = self.features(x)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/pooling.py", line 162, in forward
    return F.max_pool2d(input, self.kernel_size, self.stride,
  File "/usr/local/lib/python3.8/dist-packages/torch/_jit_internal.py", line 422, in fn
    return if_false(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/functional.py", line 797, in _max_pool2d
    return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
RuntimeError: CUDA out of memory. Tried to allocate 2.99 GiB (GPU 0; 14.76 GiB total capacity; 9.02 GiB already allocated; 2.34 GiB free; 11.18 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

The whole process took 37.50939106941223 seconds
Read 259.5968008041382 MBs for this batch
Streaming inaturalist data took 5.883686780929565 seconds
